planner/core/logical_plan_builder.go

```diff
@@ -2541,6 +2566,19 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		return nil, err
 	}
 
+	// Try to substitute generate column only if there is an index on generate column
+	for _, index := range tableInfo.Indices {
+		if index.State == model.StatePublic {
```

Comment:
How about:

```
if index.State != model.StatePublic {
    continue;
}
...
```
This style is preferred in https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting.